File: SpringTILCode/src/main/java/com/example/springtilcode/domain/user/entity/RefreshToken.java
package com.example.springtilcode.domain.user.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

import javax.persistence.Id;

@Getter
@AllArgsConstructor
@RedisHash
@Builder
public class RefreshToken {
    @Id
    private String id;

    @Indexed
    private String email;

    @TimeToLive
    private Long expiration;
}

File: kotlin/src/main/kotlin/com/example/demo/domain/token/entity/RefreshToken.kt
package com.example.demo.domain.token.entity

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.TimeToLive
import org.springframework.data.redis.core.index.Indexed

@RedisHash
data class RefreshToken (
    @Id
    val token: String?,

    @Indexed
    val email: String?,

    @TimeToLive
    val expiration: Long?
)